src/com/company/IntListImpl.java

```diff
@@ -0,0 +1,220 @@
+package com.company;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class IntListImpl<E> implements IntList{
+
+    private Integer[] array = new Integer[10];
+    private int size = size();
+
+    public Integer[] generateArray() { // создаю публичный т.к. хочу сделать проверку в мейне.
+        Integer[] arr = generateRandomArray();
+        return arr;
+    }
+
+    private static Integer[] generateRandomArray() {
+        java.util.Random random = new java.util.Random();
+        Integer[] arr = new Integer[100_000];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = random.nextInt(100_000) + 100_000;
+        }
+        return arr;
+    }
+
+    @Override
+    public Integer add(Integer item) {
+        if (size == array.length) {
+            return Integer.parseInt("Массив уже заполнен элементами");
+        }
+        array[size] = item;
+        size++;
+        return array[size];
+    }
+
+    @Override
+    public Integer add(int index, Integer item) {
+        if (index >= array.length) {
+            throw new IndexLargerSizeOfArrayException("Значение индекса " + index + " больше чем размер массива!");
+        } else {
+            array[index] = item;
+            size++;
+        }
+        return item;
+    }
+
+    @Override
+    public Integer set(int index, Integer item) {
+        if (index >= array.length) {
+            throw new IndexLargerSizeOfArrayException("Значение индекса " + index + " больше чем размер массива!");
+        } else if (array[index] == 0) {
+            throw new NullElementInTheArray("В ячейке " + index + " отсутсвует оъект, примените метод add()");
+        } else {
+            array[index] = item;
+        }
+        return item;
+    }
+
+    @Override
+    public Integer removeIt(Integer item) {
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == item) {
+                array[i] = 0;
+                size--;
+                return Integer.parseInt(item + ": Удален");
+            }
+        }
+        throw new NullElementInTheArray(item + ": данный объект не найден");
+    }
+
+    @Override
+    public Integer remove(int index) {
+        int number;
+        if (array[index] == 0) {
+            throw new NullElementInTheArray("В ячейке под индесом " + index + " null");
+        } else {
+            number = array[index];
+            array[index] = 0;
+            size--;
+        }
+        return Integer.parseInt("Элемент: " + number + " под индексом " + index + " удален");
+    }
+
+    @Override
+    public boolean contains(Integer item) {
+        Integer[] arr = array;
```

Comment:
```suggestion
        Integer[] arr = Arrays.copyOf(array, size);
```
иначе мы все еще изменяем исходный массив